packages/cysync-core/src/dialogs/AddAccountGuide/Dialogs/Device/AddAccountSingleChainDialog.tsx

```diff
@@ -0,0 +1,254 @@
+  {
+    id: '2',
+    leftImageSrc: bitcoinIcon,
+    text: 'Bitcoin 2',
+    checkType: 'checkbox',
+    tag: 'TAPROOT',
+  },
+  {
+    id: '3',
+    leftImageSrc: bitcoinIcon,
+    text: 'Bitcoin 2',
+    checkType: 'checkbox',
+    tag: 'SEGWIT',
+  },
+  {
+    id: '4',
+    leftImageSrc: bitcoinIcon,
+    text: 'Bitcoin 2',
+    checkType: 'checkbox',
+    tag: 'NATIVE SEGWIT',
+  },
+];
+const accountsInPortfolio = [
+  {
+    id: '11',
+    leftImageSrc: bitcoinIcon,
+    text: 'Bitcoin 1',
+    checkType: 'checkbox',
+    tag: 'SEGWIT',
+  },
+  {
+    id: '12',
+    leftImageSrc: bitcoinIcon,
+    text: 'Bitcoin 1',
+    checkType: 'checkbox',
+    tag: 'NATIVE SEGWIT',
+  },
+];
+
+export const AddAccountSingleChainDialog: FC = () => {
+  const [isChecked, setIsChecked] = useState(false);
+  const { goTo } = useAddAccountDialog();
+  const [checkedItems, setCheckedItems] = useState<string[]>([]);
+
+  const handleCheckBoxChange = (id: string) => {
+    if (checkedItems.includes(id)) {
+      setCheckedItems(prevItems => prevItems.filter(item => item !== id));
+    } else {
+      setCheckedItems(prevItems => [...prevItems, id]);
+    }
+  };
+
+  const handleToggleChange = (checked: boolean) => {
+    setIsChecked(checked);
+  };
+
+  const handleDeselectAll = () => {
+    const uncheckedItems = accountNotSynced.map(data => data.id);
+    setCheckedItems(prevItems =>
+      prevItems.filter(item => !uncheckedItems.includes(item)),
+    );
+  };
+
+  const checkedItemsCount = checkedItems.filter(item =>
+    accountNotSynced.some(data => data.id === item),
+  ).length;
+
+  const handleButtonClick = () => {
+    goTo(2, 0);
+  };
+
+  const lang = useAppSelector(selectLanguage);
+  const singleChain =
+    lang.strings.addAccount.addAccount.addAccountSingleChain.info.dialogBox;
+
+  return (
+    <DialogBox width={500}>
+      <DialogBoxHeader height={56} width={500}>
+        <Typography variant="fineprint" width="100%" color="muted">
+          <LangDisplay text={singleChain.title} />
+        </Typography>
+      </DialogBoxHeader>
+      <FlexGapContainer pt={4} pr={5} pl={5}>
+        <Image src={settingsIcon} alt="Loader" />
+        <Typography variant="h5" $textAlign="center">
+          <LangDisplay text={singleChain.header} />
+        </Typography>
+      </FlexGapContainer>
+      <DialogBoxBody py={4} px={5}>
+        <Flex direction="column">
+          <InputLabel
+            mt={4}
+            mr={1}
+            mb={1}
+            display="inline-block"
+            $fontSize={14}
+            $fontWeight="normal"
+          >
+            <LangDisplay text={singleChain.subheader} />
+          </InputLabel>
+          <LeanBoxContainer>
+            {dataArray.map(data => (
+              <LeanBox
+                key={data.id}
+                leftImageSrc={data.leftImageSrc}
+                text={data.text}
+                tag={data.tag}
+                {...(data.checkType ? { checkType: 'checkbox' } : {})}
```

Comment:
Why can't we just do this
```suggestion
                checkType={data.checkType ?? undefined}
```

Reply:
It was giving this error, hence as a work around used that
`
@cypherock/cysync-core:build: src/dialogs/AddAccountGuide/Dialogs/Device/AddAccountSingleChainDialog.tsx(149,17): error TS2322: Type 'string' is not assignable to type '"radio" | "checkbox" | undefined'.`

alternatively I can do something like this:
`    checkType={data.checkType as 'checkbox' | 'radio' | undefined}`